File: Rinnerdos-cs50-problems-2020-x-sentimental-credit/credit.py
from cs50 import get_string

ccNumb = get_string("Enter cc-number: \n")

count = 0
sum1 = 0

# First sum the to be multiplied digits
for i in range(len(ccNumb)-2, -1, -2):
    count += 1
    number1 = int(ccNumb[i])*2
    
    # If the to be multiplied number is > 10, add digits seperately
    if number1 >= 10:
        sum1 += 1 + number1 % 10
    else:
        sum1 += number1
  
sum2 = 0

# Then sum the non-multiplied (remainging) digits
for i in range(len(ccNumb)-1, -1, -2):
    count += 1
    sum2 += int(ccNumb[i])
    
sumTot = sum1 + sum2

# Then check which company cc number belongs to, or if invalid
if sumTot % 10 == 0:
    if (count == 15 and (ccNumb[0] == "3" and (ccNumb[1] == "4" or "7"))):
        print("AMEX")
    elif (count == 16 and (ccNumb[0] == "5" and (ccNumb[1] == "1" or "2" or "3" or "4" or "5"))):
        print("MASTERCARD")
    elif (count == 13 or 16) and ccNumb[0] == "4":
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: Rinnerdos-cs50-problems-2020-x-credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number_cc = get_long("Enter cc-number: \n");

    long count = 0;
    
    long n = number_cc;
    // This gets the length of the number
    while (n != 0)
    {
        n /= 10;
        count++;
    }
    // First sum the to be multiplied numbers
    long number1 = number_cc/10;
    long sum1 = 0;
    long digit1;
    do
    {
        digit1 = number1%10;
        number1 /= 100;
        long multi = digit1*2;

        if (multi >= 10)
        {
            sum1 += 1 + (multi%10);
        }
        else
        {
            sum1 += multi;
        }
    } while (number1 != 0);
    //Now sum the non multiplied numbers
    long number2 = number_cc;
    long sum2 = 0;
    long digit2;
    do
    {
        digit2 = number2%10;
        number2 /= 100;
        sum2 += digit2;
    } while (number2 != 0);
    long sum_tot = sum1 + sum2;
    
    // Check for sum
    if(sum_tot % 10 == 0)
    {
        // Check which company
        if (count == 15 && ((digit2 == 3 && digit1 == 4) || (digit2 == 3 && digit1 == 7)))
        {    
            printf("AMEX\n");
        }
        else if (count == 16 && (digit1 == 5 && (digit2 == 1 || digit2 == 2 || digit2 == 3 || digit2 == 4 || digit2 == 5)))
        {
            printf("MASTERCARD\n");
        }
        else if ((count == 13 || count == 16) && (digit1 == 4))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    // Else, invalid
    else
    {
        printf("INVALID\n");
    }
}
